frontend/src/scenes/insights/Histogram/Histogram.tsx

```diff
@@ -0,0 +1,168 @@
+import React from 'react'
+import * as d3 from 'd3'
+import { D3Selector, useD3, getOrCreateEl } from 'lib/hooks/useD3'
+import { FunnelLayout } from 'lib/constants'
+import { getChartColors } from 'lib/colors'
+import { getConfig, createRoundedRectPath } from './histogramUtils'
+
+import './Histogram.scss'
+import { humanFriendlyDuration } from 'lib/utils'
+
+interface HistogramDatum {
+    id: string | number
+    bin0: number
+    bin1: number
+    count: number
+}
+
+interface HistogramProps {
+    data: HistogramDatum[]
+    layout?: FunnelLayout
+    color?: string
+}
+
+export function Histogram({ data, layout = FunnelLayout.vertical, color = 'white' }: HistogramProps): JSX.Element {
+    const colorList = getChartColors(color)
+
+    // Initial dimensions
+    const isVertical = layout === FunnelLayout.vertical
+    const config = getConfig(isVertical)
+
+    console.log('CONFIG', config)
+
+    const ref = useD3(
+        (container) => {
+            // x-axis scale
+            const xMax = data[data.length - 1].bin1
+            const x = d3.scaleLinear().domain([data[0].bin0, xMax]).range(config.ranges.x).nice()
+            const xAxis = config.axisFn
+                .x(x)
+                .tickValues([...data.map((d) => d.bin0), xMax])
+                // v === -2 || v === -1 represent bins that catch grouped outliers.
+                // TODO: (-2, -1) are temporary placeholders for (-inf, +inf) and should be changed when backend specs are finalized
+                .tickFormat((v: number) => {
+                    const label = humanFriendlyDuration(v)
+                    if (v === -2) {
+                        return `<${label}`
+                    }
+                    if (v === -1) {
+                        return `>=${label}`
+                    }
+                    return label
+                })
+
+            // y-axis scale
+            const y = d3
+                .scaleLinear()
+                .domain([0, d3.max(data, (d: HistogramDatum) => d.count) as number])
+                .range(config.ranges.y)
+                .nice()
+            const yAxis = config.axisFn.y(y).tickSize(0)
+
+            // y-axis gridline scale
+            const yAxisGrid = config.axisFn
+                .y(y)
+                .tickSize(-config.gridlineTickSize)
+                .tickFormat('')
+                .ticks(y.ticks().length)
+
+            const renderCanvas = (parentNode: D3Selector): D3Selector => {
+                // Get or create svg > g
+                const _svg = getOrCreateEl(parentNode, 'svg > g', () =>
+                    parentNode
+                        .append('svg:svg')
+                        .attr('viewBox', `0 0 ${config.width} ${config.height}`)
+                        .append('svg:g')
+                        .classed(isVertical ? FunnelLayout.vertical : FunnelLayout.horizontal, true)
```

Comment:
nit: couldn't this be `.classed(layout, true)` to better match the search `.classed(layout)` below?

Reply:
very true 👍